usaspending_api/agency/v2/views/agency_base.py

```diff
@@ -78,6 +83,36 @@ def validate_fiscal_period(request_data):
         if fiscal_period < 2 or fiscal_period > 12:
             raise UnprocessableEntityException(f"fiscal_period must be in the range 2-12")
 
+    @staticmethod
+    def create_assurance_statement_url(result):
+        """
+        Results requires the following keys to generate the assurance statement url:
+        agency_name, abbreviation, toptier_code, fiscal_year, fiscal_period, submission_is_quarter
+        """
+        try:
+            agency_name_split = result["agency_name"].split(" ")
+            abbreviation_wrapped = f"({result['abbreviation']})"
+            toptier_code = result["toptier_code"]
+            fiscal_year = result["fiscal_year"]
+
+            if result["submission_is_quarter"]:
+                fiscal_period = f"Q{int(result['fiscal_period'] / 3)}"
```

Comment:
We have a `get_quarter_from_period()` method in fiscal_year_helpers, which could be useful here.

Reply:
The method you listed did not meet the needs here exactly, so I left it as is

Reply:
fair enough